libpdbg/device.c

```diff
@@ -48,11 +48,119 @@ static const char *take_name(const char *name)
 	return name;
 }
 
+static const char *get_unitname(const struct pdbg_target *node)
+{
+	const char *c = strchr(node->dn_name, '@');
+
+	if (!c)
+		return NULL;
+
+	return c + 1;
+}
+
+/* The moment any number is found, in a string:
+   separate the string into number and string without numbers.
+   Return the number found, update the string passed to chop the number from it
+   Find only the number at the end of the string*/
+unsigned int strip_num_at_the_end(char *str) 
+{
+    unsigned int num = 0;
+    unsigned int only_number = 0;
+    int len = strlen(str);
+    int last_index_of_char = len;
+
+    for(int i = 0; i < len; i++) {
+        if(isdigit(str[i])) {
+            num = num * 10 + (str[i] - '0');
+            /*comes here at the end for the null character*/
+            if(!isdigit(str[i+1])) {
+                only_number = num;
+                num = 0;
+            }
+        }
+        else {
+         last_index_of_char = i;
+        }
+    }
+    
+    str[last_index_of_char] = '\0';
+    return only_number;
+}
+
+static int dt_cmp_subnodes(const struct pdbg_target *a, const struct pdbg_target *b)
```

Comment:
Can we have better param names for a and b? Also a bit of doxygen comment about function description would be nice to have :-)

Reply:
This is an existing function. I will surely add the function description. But may not change the name as a and b are are defined just at the start of the function and we understand that we are comparing them.